scripts/publish_example_server_container.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+[ -z "$CI" ] && echo "This script is meant to run only from CircleCI." && exit 1;
+[ -z "$DOCKERHUB_USERNAME" ] && echo "Undefined DOCKERHUB_USERNAME, skipping publish" && exit 0;
+
+docker login -u $DOCKERHUB_USERNAME -p $DOCKERHUB_PASSWORD
+
+#use TAG env. variable to create the container with the given tag
+TAG="${TAG:-latest}"
+
+NAMESPACE="aerogear"
+CONTAINER="$NAMESPACE/voyager-server-example-task:$TAG"
+CONTAINER_LATEST="$NAMESPACE/voyager-server-example-task:latest"
+
+echo "Building docker container $CONTAINER"
+docker build -f Dockerfile -t "$CONTAINER" "$WORKDIR" && docker push "$CONTAINER" && docker push "$CONTAINER_LATEST"
```

Comment:
IMO, it's much better to put these statements on their own lines. There is no need to group them together because failure in any of them will fail the build. It makes the line harder to read and during failures in CI it makes the error output more difficult to debug because you're wondering which statement failed.

Additionally, the `docker push $CONTAINER_LATEST` statement will likely fail here because you need to tag the image. You need to do something like `docker build -f Dockerfile -t $CONTAINER -t CONTAINER_LATEST` that will ensure the image is tagged with both the version tag and with `latest`.

Reply:
There is `docker tag` command. It will add new tag there. No need to build it again.

Reply:
Sorry I added it after you added your comments.